server/src/lib/server.rs

```diff
@@ -259,61 +259,17 @@ async fn instance_ensure(
         state: propolis::instance::State::Initialize,
     });
 
-    instance.on_transition(Box::new(move |next_state, _inv, ctx| {
-        match next_state {
-            propolis::instance::State::Boot => {
-                // Set vCPUs to their proper boot (INIT) state
-                for mut vcpu in ctx.mctx.vcpus() {
-                    vcpu.reboot_state().unwrap();
-                    vcpu.activate().unwrap();
-                    // Set BSP to start up
-                    if vcpu.is_bsp() {
-                        vcpu.set_run_state(bhyve_api::VRS_RUN).unwrap();
-                        vcpu.set_reg(
-                            bhyve_api::vm_reg_name::VM_REG_GUEST_RIP,
-                            0xfff0,
-                        )
-                        .unwrap();
-                    }
-                }
-            }
-            _ => {}
-        }
+    instance.on_transition(Box::new(move |next_state, _inv, _ctx| {
         let last = (*tx.borrow()).clone();
         let _ = tx.send(StateChange { gen: last.gen + 1, state: next_state });
```

Comment:
This will mean that emitting the `StateChange` objects through the channel will now occur before the later `on_transition` event performing the on-reboot reinitinialization of the vCPU state.  That's probably fine, though.